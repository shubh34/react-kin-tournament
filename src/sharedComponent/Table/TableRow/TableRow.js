import './TableRow.scss';

import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const TableRow = ({ children, isActive = false }) => {
	const className = classNames('c-table-row',
		{
			'c-table-row--active': isActive,
		});
	return (
		<tr className={className}>
			{children}
		</tr>
	);
};


TableRow.propTypes = {
	children: PropTypes.node.isRequired,
	isActive: PropTypes.bool,
};

export default TableRow;
