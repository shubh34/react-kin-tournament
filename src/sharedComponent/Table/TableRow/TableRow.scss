@import '../../../colors';
.c-table-row {
    text-align: center;
    line-height: 80px;
    background: $lightGray;
    & :first-child {
        background: transparent;
    }
    th, td {
        border: none;
        &:first-child {
            float: left;
        } 
        &:last-child {
            float: right;
        }
    }
    td {
        border: none;
    }
}

.c-table-row--header {
   background: transparent;
}
.c-table-row--active {
    background: $green;
    color: $lightGray
 }
 
 .c-table-row :first {

    background: transparent;
 }