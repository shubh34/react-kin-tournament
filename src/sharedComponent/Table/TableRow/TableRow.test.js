import React from 'react';
import { shallow } from 'enzyme';
import TableRow from './TableRow';

describe('TableRow', () => {
	it('renders without crashing', () => {
		const wrapper = shallow(<TableRow><td>td </td></TableRow>);
		expect(wrapper).toMatchSnapshot();
	});
	it('renders the active class', () => {
		const wrapper = shallow(<TableRow isActive><td>td </td></TableRow>);
		expect(wrapper).toMatchSnapshot();
	});
});
