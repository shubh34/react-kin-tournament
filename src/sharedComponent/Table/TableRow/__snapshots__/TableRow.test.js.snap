// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableRow renders the active class 1`] = `
<tr
  className="c-table-row c-table-row--active"
>
  <td>
    td 
  </td>
</tr>
`;

exports[`TableRow renders without crashing 1`] = `
<tr
  className="c-table-row"
>
  <td>
    td 
  </td>
</tr>
`;
